File: uniffi_bindgen/src/bindings/python/templates/UInt8Helper.py
class FfiConverterUInt8(FfiConverterPrimitive):
    @classmethod
    def lower(cls, value):
        if not 0 <= value < 2**8:
            raise ValueError("u8 requires {} <= value < {}".format(0, 2**8))
        return super().lower(value)

    @staticmethod
    def read(buf):
        return buf.readU8()

    @staticmethod
    def write(value, buf):
        buf.writeU8(value)

File: uniffi_bindgen/src/bindings/swift/templates/UInt8Helper.swift
fileprivate struct FfiConverterUInt8: FfiConverterPrimitive {
    typealias FfiType = UInt8
    typealias SwiftType = UInt8

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> UInt8 {
        return try lift(readInt(&buf))
    }

    static func write(_ value: UInt8, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}
